ibis/expr/types/arrays.py

```diff
@@ -1059,6 +1059,220 @@ def flatten(self) -> ir.ArrayValue:
         """
         return ops.ArrayFlatten(self).to_expr()
 
+    def any(self) -> ir.BooleanValue:
+        """Return whether any element in the array is true.
+
+        Returns
+        -------
+        BooleanValue
+            Whether any element in the array is true
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable(
+        ...     {
+        ...         "arr": [
+        ...             [True, False],
+        ...             [False],
+        ...             [True],
+        ...             [None, False],
+        ...             [None, True],
+        ...             [None],
+        ...             [],
+        ...             None,
+        ...         ]
+        ...     }
+        ... )
+        >>> t.mutate(x=t.arr.any())
+        ┏━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━┓
+        ┃ arr                  ┃ x       ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━┩
+        │ array<boolean>       │ boolean │
+        ├──────────────────────┼─────────┤
+        │ [True, False]        │ True    │
+        │ [False]              │ False   │
+        │ [True]               │ True    │
+        │ [None, False]        │ False   │
+        │ [None, True]         │ True    │
+        │ [None]               │ NULL    │
+        │ []                   │ NULL    │
+        │ NULL                 │ NULL    │
+        └──────────────────────┴─────────┘
+        """
+        return ops.ArrayAny(self).to_expr()
+
+    def all(self) -> ir.BooleanValue:
+        """Return whether all elements in the array are true (ignoring nulls).
+
+        Returns
+        -------
+        BooleanValue
+            Whether all elements in the array are true (ignoring nulls).
+            Will be NULL if no elements are non-NULL.
```

Comment:
```suggestion
            Will be NULL if all elements are NULL.
```

Seems a bit easier to grok IMO.

Reply:
That was purposefull so empty arrays are included, though I agree that ALSO is subtle. Maybe "all are NULL or empty" and just be explicit?

Reply:
I went with `all are NULL or empty`